src/npem.c

```diff
@@ -182,14 +199,19 @@ static int npem_wait_command(struct pci_dev *pdev)
 int npem_write(struct block_device *device, enum ibpi_pattern ibpi)
 {
 	struct cntrl_device *npem_cntrl = device->cntrl;
-	struct pci_access *pacc = NULL;
 	struct pci_dev *pdev = NULL;
+	struct pci_access *pacc = get_pci_access();
 
 	u32 reg;
 	u32 val;
 
 	int err = 0;
 
+	if (!pacc) {
```

Comment:
Can we fall back to initial version then? I don't see any reason to change pacc handling in npem_write